Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs

```diff
@@ -486,80 +486,128 @@ public IEnumerator ManipulationHandlerOneHandMoveFar()
+                float diffAngle = Quaternion.Angle(testObject.transform.rotation, Quaternion.Euler(testRotation, testRotation, testRotation) * initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "object didn't rotate with hand");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "object didn't rotate with hand");
+
+                // rotate with x axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.XAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, Quaternion.Euler(testRotation, 0, 0) * initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly");
+
+                // rotate with y axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.YAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, Quaternion.Euler(0, testRotation, 0) * initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly");
+
+                // rotate with z axis only
+                manipHandler.ConstraintOnRotation = RotationConstraintType.ZAxisOnly;
+                yield return hand.SetRotation(testQuaternion, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, Quaternion.Euler(0, 0, testRotation) * initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly");
+
+                yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+                diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+                Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly");
+
+                // Switch to local rotation
+                manipHandler.UseLocalSpaceForConstraint = true;
```

Comment:
Would prefer to have two tests: one for local space constratint, one for world space constraint. Will make it much easier to understand test failures.

Reply:
Yeah, I am open to creating a second. Maybe I should just elaborate on the descriptions of the failures. Thoughts?

Reply:
Kept single methods, added more comments.